Data/ByteString/Internal.hs-boot

```diff
@@ -0,0 +1,11 @@
+-- allow Data.ByteString.Internal.Utils to use the BS constructor
+
+module Data.ByteString.Internal (
```

Comment:
I'm cautious about `hs-boot` files. Let's make it `Data.ByteString.Internal.Type` or similar.

Reply:
OK, I have now tried just moving the module wholesale and leaving a re-export compatibility stub in its place. Unfortunately this makes the displayed diff for the PR as a whole much uglier.